packages/@sanity/base/src/components/previews/general/DetailPreview.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+import {Box, Text, Skeleton, TextSkeleton, useTheme, rem} from '@sanity/ui'
+import {getDevicePixelRatio} from 'use-device-pixel-ratio'
+import styled from 'styled-components'
+import {Media} from '../_common/Media'
+import {PreviewMediaDimensions, PreviewProps} from '../types'
+import {PREVIEW_MEDIA_SIZE} from '../constants'
+import {Root, HeaderFlex, Content, HeaderStack, StatusBox} from './DetailPreview.styled'
+
+export type DetailPreviewProps = PreviewProps<'detail'>
+
+const DEFAULT_MEDIA_DIMENSIONS: PreviewMediaDimensions = {
+  ...PREVIEW_MEDIA_SIZE.detail,
+  fit: 'crop',
+  aspect: 1,
+  dpr: getDevicePixelRatio(),
+}
+
+const MediaSkeleton = styled(Skeleton).attrs({animated: true, radius: 2})`
```

Comment:
Maybe these skeleton components should be moved into `DetailPreview.styled` – so that all styling is gathered in one place?